pkg/ingestor/verifier/verifier.go

```diff
@@ -42,6 +49,29 @@ type Verifier interface {
 // way.
 type Identity struct {
 	ID       string
-	Key      *crypto.PublicKey
+	Key      key.Key
 	Verified bool
 }
+
+var (
+	verifierProviders = map[VerifierType]Verifier{}
+)
+
+// RegisterVerifier registers the providers that are available for verification
+func RegisterVerifier(k Verifier, providerType VerifierType) {
+	if _, ok := verifierProviders[providerType]; ok {
+		logrus.Warnf("the verification provider is being overwritten: %s", providerType)
```

Comment:
we changed this in the logger PR to just return an error instead.

Reply:
yup. changed.